.dotnet/src/Custom/Assistants/AssistantClient.Convenience.cs

```diff
@@ -123,64 +122,40 @@ public virtual ClientResult<bool> DeleteThread(AssistantThread thread)
             => CreateMessage(thread?.Id, content, options);
 
     /// <summary>
-    /// Returns a list of <see cref="ThreadMessage"/> instances from an existing <see cref="AssistantThread"/>,
-    /// matching any optional constraints provided.
+    /// Returns a collection of <see cref="ThreadMessage"/> instances from an existing <see cref="AssistantThread"/>.
     /// </summary>
     /// <param name="thread"> The thread to list messages from. </param>
-    /// <param name="maxResults">
-    /// A <c>limit</c> for the number of results in the list. Valid in the range of 1 to 100 with
-    /// a default of 20 if not otherwise specified.
-    /// </param>
     /// <param name="resultOrder">
     /// The <c>order</c> that results should appear in the list according to their <c>created_at</c>
     /// timestamp.
     /// </param>
-    /// <param name="previousId">
-    /// A cursor for use in pagination. If provided, results in the list will begin immediately
-    /// <c>after</c> this ID according to the specified order.
-    /// </param>
-    /// <param name="subsequentId">
-    /// A cursor for use in pagination. If provided, results in the list will end just <c>before</c>
-    /// this ID according to the specified order.
-    /// </param>
-    /// <returns> A page of results matching any constraints provided. </returns>
-    public virtual ClientResult<ListQueryPage<ThreadMessage>> GetMessages(
+    /// <returns> A collection of messages that can be enumerated using <c>await foreach</c>. </returns>
+    public virtual AsyncPageableCollection<ThreadMessage> GetMessagesAsync(
         AssistantThread thread,
-        int? maxResults = null,
-        ListOrder? resultOrder = null,
-        string previousId = null,
-        string subsequentId = null)
-            => GetMessages(thread?.Id, maxResults, resultOrder, previousId, subsequentId);
+        ListOrder? resultOrder = default)
```

Comment:
I think we may want to retain the `after` and `before` as optional inputs into the public surface; uses like "get all the messages created after this one" or "start listing the assistants that were created before my known one" still have validity.

Reply:
I had understood that `after` and `before` were just pagination cursors to ensure callers got the net page of results and not pages of repeated values.  If this is correct, we expose the ability to use those via the `AsPages` APIs on the PageableCollection types.  Let me know if you feel strongly that these are part of a higher-level convenience story, though!

Reply:
I'll add a test for paging in reverse order and using `before`, though ... that may not be correctionally functional at present.

Reply:
Ah, actually it looks like `FunctionToolsWork` is already using ListOrder.NewestFirst.

Reply:
Thinking on this more: I do suspect that we'll still end up with the request to expose the pagination cursors to avoid a bunch of extra pages, but it'd also be possible to add (more parameters) as a non-breaking change. So this doesn't need to be blocking one way or another, even for GA.

Reply:
Great!  Yep, if customers ask for them, they are entirely additive, and we will know at those times the precise scenarios they are looking to enable, so we can design the best thing for those use cases.  Thanks!